enhancements/oc/day2-add-nodes.md

```diff
@@ -0,0 +1,271 @@
+
+There are several reasons for choosing `oc` as a primary user interfaces for these tasks:
+- It’s the official cluster administration tool, so it is expected to be already available in the target cluster environment
+- It already supports the kubeconfig file for interacting with the target cluster (it’s relevant since it contains several pieces of 
+  information required for adding the nodes, as explained in the design section)
+- It already offers a couple of commands to manually approve/deny certificate signing requests (CSRs) when adding new nodes
+
+The usage of `oc` will also help in reducing the amount of information that the user will need to provide, resulting thus in a simplified
+input configuration file.
+
+### add-nodes-image command
+The format of the command will be the following:
+
+```
+oc --kubeconfig='' adm add-nodes-image [-o='agent-nodes.iso'] <nodes-config.yaml> 
+```
+
+Where:
+- `-o='agent-nodes.iso'`. The name of the ISO image to be generated.
+- `--kubeconfig=''`. Path to the kubeconfig file (or `KUBECONFIG` if available)
+
+`nodes-config.yaml` is a configuration file, similar to `agent-config.yaml`, used to specify the list of the hosts to be added as worker 
+nodes. For each host, it will be possible to specify a number of attributes, including a piece of nmstate configuration for the initial 
+static network configuration. If not specified, the tool will look for a `nodes-config.yaml` file in the current directory.
+
+The command will use the configured kubeconfig file, and the connection to the target cluster, to retrieve all the additional information 
+required to validate and generate the ISO image that will be used for adding the configured nodes. 
+
+#### nodes-config.yaml sample file
+Following an example of `nodes-config.yaml` file, where a couple of workers are described with an initial static network configuration:
+
+```
+hosts:
+- hostname: extra-worker-1
+  rootDeviceHints:
+   deviceName: /dev/sda
+  interfaces:
+   - macAddress: 00:00:00:00:00:00
+     name: eth0
+   - macAddress: 00:00:00:00:00:01
+     name: eth1
+  networkConfig:
+   interfaces:
+     - name: eth0
+       type: ethernet
+       state: up
+       mac-address: 00:00:00:00:00:00
+       ipv4:
+         enabled: true
+         address:
+           - ip: 192.168.122.2
+             prefix-length: 23
+         dhcp: false
+- hostname: extra-worker-2
+  rootDeviceHints:
+   deviceName: /dev/sda
+  interfaces:
+   - macAddress: 00:00:00:00:00:02
+     name: eth0
+  networkConfig:
+   interfaces:
+     - name: eth0
+       type: ethernet
+       state: up
+       mac-address: 00:00:00:00:00:02
+       ipv4:
+         enabled: true
+         address:
+           - ip: 192.168.122.3
+             prefix-length: 23
+         dhcp: false
+```
+
+### monitor-add-nodes command
+
+The format of the command will be the following:
+
+`oc --kubeconfig='' adm monitor-add-nodes <IPs list>`
```

Comment:
One complication I could see here is if users used dhcp in their nmstate configs and if they have no way of knowing the IPs that will be assigned.

Reply:
They don't have to run this command until after the host is booted, so they should be able to know everything they are going to know at that point.